### ISSUE TITLE
snapshots not being cleaned on dest

### ISSUE THREAD

**Issue by bleomycin** [yboetz/pyznap#7]
Thanks for the great script! I'm probably doing something wrong but snapshots are not being cleaned on my destination, they're being cleaned just fine on the source datasets though. Below is my super simple pyznap.conf and a snippet from zfs list -t snapshot. This is on a proxmox host (debian stretch) with python 3.5.3.

`[p3600]
hourly = 24
daily = 7
weekly = 1
snap = yes
clean = yes
dest = tank/p3600backup

[tank/Media]
daily = 7
weekly = 1
snap = yes
clean = yes

[tank/backup]
daily = 7
weekly = 1
snap = yes
clean = yes`

`tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-16_22:13:18_weekly                 0B      -  32.0G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-16_22:13:18_daily                  0B      -  32.0G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-16_22:13:18_hourly                 0B      -  32.0G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_12:15:56_daily                  0B      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_12:15:56_hourly                 0B      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_14:00:01_hourly              17.2M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_15:00:01_hourly              15.2M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_16:00:01_hourly              3.64M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_17:00:01_hourly              3.83M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_18:00:01_hourly              13.4M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_19:00:01_hourly              14.9M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_20:00:02_hourly              15.9M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_21:00:01_hourly              16.2M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_22:00:01_hourly              19.8M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-17_23:00:02_hourly              31.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_00:00:01_daily               31.4M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_01:00:02_hourly              25.7M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_02:00:01_hourly              7.55M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_03:00:02_hourly              7.18M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_04:00:01_hourly              14.5M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_05:00:01_hourly              14.1M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_06:00:01_hourly              12.2M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_07:00:01_hourly              11.9M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_08:00:01_hourly              11.9M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_09:00:01_hourly              11.8M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_10:00:02_hourly              11.8M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_11:00:01_hourly              11.9M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_12:00:01_hourly              11.9M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_13:00:02_hourly              11.8M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_14:00:01_hourly              11.9M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_15:00:01_hourly              11.9M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_16:00:02_hourly              12.0M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_17:00:01_hourly              25.8M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_18:00:01_hourly              37.8M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_19:00:01_hourly              44.3M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_20:00:01_hourly              38.5M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_21:00:01_hourly              32.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_22:00:01_hourly              22.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-18_23:00:02_hourly              16.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_00:00:01_weekly                 0B      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_00:00:01_daily                  0B      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_01:00:01_hourly              14.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_02:00:02_hourly              15.4M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_03:00:01_hourly              16.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_04:00:02_hourly              12.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_05:00:02_hourly              11.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_06:00:01_hourly              12.0M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_07:00:01_hourly              11.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_08:00:01_hourly              12.0M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_09:00:01_hourly              11.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_10:00:01_hourly              11.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_11:00:02_hourly              11.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_12:00:01_hourly              11.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_13:00:02_hourly              11.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_14:00:01_hourly              12.0M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_15:00:01_hourly              11.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_16:00:02_hourly              11.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_17:00:01_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_18:00:02_hourly              17.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_19:00:01_hourly              22.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_20:00:02_hourly              27.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_21:00:01_hourly              32.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_22:00:01_hourly              27.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-19_23:00:01_hourly              19.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_00:00:01_daily               16.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_01:00:01_hourly              19.6M      -  32.5G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_02:00:01_hourly              18.4M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_03:00:01_hourly              42.1M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_04:00:01_hourly              17.4M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_05:00:01_hourly              12.1M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_06:00:01_hourly              11.9M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_07:00:01_hourly              17.9M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_08:00:01_hourly              17.8M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_09:00:01_hourly              3.31M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_10:00:01_hourly              2.60M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_11:00:01_hourly              21.8M      -  32.6G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_12:00:01_hourly              12.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_13:00:01_hourly              12.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_14:00:01_hourly              21.6M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_15:00:01_hourly              17.5M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_16:00:01_hourly              12.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_17:00:02_hourly              12.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_18:00:01_hourly              15.3M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_19:00:02_hourly              25.5M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_20:00:01_hourly              27.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_21:00:02_hourly              21.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_22:00:01_hourly              8.13M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-20_23:00:01_hourly              3.25M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_00:00:01_daily               12.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_01:00:01_hourly              12.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_02:00:01_hourly              17.6M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_03:00:01_hourly              26.9M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_04:00:02_hourly              13.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_05:00:01_hourly              12.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_06:00:01_hourly              11.5M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_07:00:01_hourly              12.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_08:00:01_hourly              31.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_09:00:01_hourly              12.4M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_10:00:01_hourly              11.9M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_11:00:01_hourly              11.8M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_12:00:01_hourly              12.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_13:00:01_hourly              12.0M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_14:00:01_hourly              12.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_15:00:02_hourly              12.2M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_16:00:01_hourly              12.2M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_17:00:01_hourly              12.9M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_18:00:01_hourly              27.6M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_19:00:01_hourly              34.1M      -  32.7G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_20:00:01_hourly              29.4M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_21:00:01_hourly              26.0M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_22:00:01_hourly              26.6M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-21_23:00:01_hourly              31.5M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_00:00:02_daily               21.7M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_01:00:01_hourly              18.1M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_02:00:01_hourly              24.4M      -  32.8G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_03:00:01_hourly              32.9M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_04:00:01_hourly              17.2M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_05:00:01_hourly              12.9M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_06:00:01_hourly              12.2M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_07:00:01_hourly              12.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_08:00:02_hourly              2.38M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_09:00:01_hourly              2.39M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_10:00:01_hourly              11.9M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_11:00:01_hourly              12.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_12:00:01_hourly              12.1M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_13:00:01_hourly              12.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_14:00:01_hourly              12.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_15:00:02_hourly              13.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_16:00:01_hourly              14.4M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_17:00:01_hourly              15.2M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_18:00:01_hourly              14.3M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_19:00:01_hourly              13.1M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_20:00:02_hourly              31.1M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_21:00:01_hourly              29.5M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_22:00:01_hourly              23.0M      -  32.1G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-22_23:00:02_hourly              15.2M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_00:00:01_daily               12.5M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_01:00:01_hourly              12.5M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_02:00:01_hourly              12.9M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_03:00:01_hourly              37.3M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_04:00:01_hourly              12.8M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_05:00:01_hourly              12.1M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_06:00:01_hourly              12.1M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_07:00:01_hourly              12.6M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_08:00:02_hourly              16.4M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_09:00:01_hourly              12.0M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_10:00:02_hourly              12.0M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_11:00:01_hourly              26.0M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_12:00:01_hourly              13.3M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_13:00:02_hourly              12.4M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_14:00:01_hourly              13.8M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_15:00:01_hourly              12.1M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_16:00:01_hourly              12.1M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_17:00:01_hourly              17.2M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_18:00:01_hourly              19.9M      -  32.2G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_19:00:01_hourly              41.3M      -  32.3G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_20:00:01_hourly              25.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_21:00:01_hourly              22.9M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_22:00:01_hourly              20.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-23_23:00:02_hourly              19.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_00:00:02_daily               17.0M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_01:00:01_hourly              15.6M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_02:00:01_hourly              15.8M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_03:00:01_hourly              20.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_04:00:01_hourly              21.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_05:00:02_hourly              12.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_06:00:01_hourly              12.2M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_07:00:01_hourly              18.7M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_08:00:01_hourly              12.5M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_09:00:02_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_10:00:01_hourly              24.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_11:00:01_hourly              13.4M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_12:00:01_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_13:00:01_hourly              12.5M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_14:00:01_hourly              14.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_15:00:01_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_16:00:02_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_17:00:01_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_18:00:01_hourly              12.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_19:00:01_hourly              14.3M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_20:00:02_hourly              15.1M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_21:00:01_hourly              15.2M      -  32.4G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_22:00:01_hourly              21.7M      -  32.5G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-24_23:00:02_hourly              17.1M      -  32.5G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-25_00:00:01_daily                  0B      -  32.5G  -
tank/p3600backup/subvol-125-disk-1@pyznap_2018-03-25_00:00:01_hourly                 0B      -  32.5G  -`

**Comment by yboetz** [comment]
Hi

You have to make a separate config entry for your dest. Reason here is that you might want to keep more/older snapshots on your backup pool and only keep a few on your working pool. So pyznap does not automatically clean dest with the same config as your source. In your case you only have to make another entry in the config with:


[tank/p3600backup]
hourly = 24
daily = 7
weekly = 1
snap = no
clean = yes


Note that on your dest pool you should have `snap = no`, as you don't want to take more snapshots of this if you're sending snapshots from some other place. If you take more snapshots then send/recv will fail due to there being more snapshots on dest compared to source. You can also keep more snapshots on dest, e.g. with `hourly = 48`, then you can go back further on your backup pool. 

Once you add this config entry and let pyznap run once all unwanted snapshots will be cleaned.

**Comment by bleomycin** [comment]
Awesome,  that did the trick! Thank you for the help, I definitely didn't understand that was how it worked before.

**Comment by netpro25** [comment]
I found this helpful, but I was looking to clean an SSH destination. I found that if I add an SSH destination as follows the snapshots are cleaned on the remote host. One caveat I found is that you cannot specify ssh_keys as it is ignored.

[ssh:22:user@host:backup/data]
hourly = 24
daily = 7
weekly = 1
snap = no
clean = yes

**Comment by yboetz** [comment]
The config entry to specify an ssh key is just called `key`. So your config would look like

    [ssh:22:user@host:backup/data]
    key = /home/user/.ssh/id_rsa
    hourly = 24
    daily = 7
    weekly = 1
    snap = no
    clean = yes

If no key is given then the standard ssh directory of the user is checked for a valid key. If there is none then that dataset will be skipped. I don't have that in the readme, only in the sample config. Maybe I should mention that in the readme as well.

If you are replicating to multiple ssh destinations then the config entry to specify those keys is called `dest_keys`, used like this:

    [tank/data]
    dest = backup/data, ssh::user0@host0:backup/data, ssh:22:user1@host1:backup/data
    dest_keys = /home/user0/.ssh/id_rsa, /home/user1/.ssh/id_rsa

**Comment by netpro25** [comment]
Yboetz,

That works perfectly! I overlooked that in the sample config. Thanks a bunch!

**Comment by florian-obradovic** [comment]
Awesome work! Love the tool!

**Comment by yboetz** [comment]
pyznap only touches snapshot it created itself, i.e. only the snapshots with the format `@pyznap_...`. So if you want to clean up old snapshots from zfs-auto-snapshot on the dest you need to do that with zfs-auto-snapshot,

**Comment by abclution** [comment]
Thanks for the reply. Yeah I figured as much. 

I may investigate switching completely over to pyznap for the samba shadow copies ( I saw a previously answered thread about it) but I will have to do double snapshots (zfs-auto-snapshot & pyznap snapshots) for a while before switching over based on shadow copies retention policies. 

To have zfs-auto-snapshot clean out the snapshots, I ended up having to temporarily enable (com.sun:auto-snapshot=true) on the destination dataset, run the to zfs-auto-snapshot
  cleanup process (zfs-auto-snapshot --destroy-only --recursive --verbose --syslog --label=weekly --keep=8  mypool)  and then disable com.sun:auto-snapshot=false afterwards. I just edited the zfs-autosnapshot cron jobs and put the needed lines there. 

Clunky but usable.
